core/domain/state_domain.py

```diff
@@ -540,6 +587,53 @@ def from_dict(cls, interaction_dict):
             [Hint.from_dict(h) for h in interaction_dict['hints']],
             solution_dict)
 
+    def to_proto(self):
+        """Returns a proto representation of the interaction object.
+
+        Returns:
+            InteractionInstance|dict. The InteractionInstance proto object
+            if the id of the interaction matches else return an empty dict.
+        """
+        interaction = None
+        interaction_proto = {}
+        if self.id is not None:
+            interaction = (
+                interaction_registry.Registry.get_interaction_by_id(
+                    self.id))
+
+        if self.id == 'Continue':
+            interaction_proto = state_pb2.InteractionInstance(
+                continue_instance=interaction.to_proto(self))
```

Comment:
Now, we are calling this `to_proto` function which is in another module, need to pass the interaction object so I am passing the self here. 
I am not sure passing self in this way is a good approach, do you have any better approach here?

Reply:
Make sure to pass something that's as minimal as possible -- self seems like it might be passing too much.